src/v/utils/vint.h

```diff
@@ -12,8 +12,109 @@
 #pragma once
 #include "bytes/bytes.h"
 
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/iostream.hh>
+
 #include <cstdint>
 
+namespace unsigned_vint {
```

Comment:
I don't know if c++ optimizes the loops below very hard, but if not, I have some optimizations in my own kbin package that may be worth stealing into this: https://github.com/twmb/franz-go/blob/master/pkg/kbin/primitives.go#L74-L181

Basically, a loop unrolled uvarint decoder, and a switch on the uvarint length for encoding so there is no looping.

Reply:
why do you need `unsigned` vint ? it is already 9 bytes, when is the kafka protocol going to need that one more bit? why not simply cast it ? 

I think today the vint serializer takes care of signs too

Reply:
so max_vint is 9 bytes, in case you use INT64_MAX

but i can't see any part of the kafka API that would possibly use a number that large. why have 2 impls. the current one already handles that correctly.

Reply:
ah i see. nvm.